src/hooks/useGetPurposeMessageKeys.ts

```diff
@@ -0,0 +1,47 @@
+import { ConsentSelection } from '../types';
+
+import { useMemo } from 'preact/hooks';
+
+import { DefinedMessage } from '@transcend-io/internationalization';
+
+export interface PurposeMessages {
+  /** The lookup of purpose trackingTypes to message key */
+  purposeToMessageKey: Record<string, DefinedMessage>;
+  /** The order of all purposes */
+  orderOfPurposes: string[]
+}
+
+export const useGetPurposeMessageKeys = ({ consentSelection, defaultPurposeToMessageKey }: {
+  /** The configured airgap purpose types */
+  consentSelection: ConsentSelection;
+  /** The lookup of messages for default purpose types */
+  defaultPurposeToMessageKey: Record<string, DefinedMessage>;
+}): PurposeMessages => {
+  const purposeToMessageKey: Record<string, DefinedMessage> = useMemo(
+    () =>
+      // the purpose type is unique for the bundle
+      Object.keys(consentSelection ?? {}).reduce(
+        (allMessages, purposeType) => {
+          if (allMessages[purposeType]) {
+            return allMessages;
+          }
+          const customPurposeMessageLabel = `purpose.${purposeType}`;
```

Comment:
purposeType is guaranteed to be unique and immutable, so I'd rather use this than the uuid